pandas/tests/dtypes/test_generic.py

```diff
@@ -38,3 +39,37 @@ def test_abc_types(self):
         assert isinstance(self.sparse_array, gt.ABCSparseArray)
         assert isinstance(self.categorical, gt.ABCCategorical)
         assert isinstance(pd.Period('2012', freq='A-DEC'), gt.ABCPeriod)
+
+
+def test_setattr_warnings():
+    # GH5904 - Suggestion: Warning for DataFrame colname-methodname clash
+    # GH7175 - GOTCHA: You can't use dot notation to add a column...
+    d = {'one': pd.Series([1., 2., 3.], index=['a', 'b', 'c']),
+         'two': pd.Series([1., 2., 3., 4.], index=['a', 'b', 'c', 'd'])}
+    df = pd.DataFrame(d)
+
+    with catch_warnings(record=True) as w:
+        # successfully add new column
```

Comment:
add a statement that the first 3 should NOT be showing warnigns (I know you are checking which is fine)